### ISSUE TITLE
ColorSelector: wrap color palette in a fieldset

### ISSUE THREAD

**Issue by retrofox** [WordPress/gutenberg#18308]
Each colour palette should be wrapped in a fieldset with a legend instead of a plain div with a span for a label, so that screen readers can read out those labels.

**Comment by tellthemachines** [comment]
This issue applies everywhere the `ColorPaletteControl` component is being used, so we need to look at changing it [in that component](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/color-palette/control.js). I'll update this issue's labels accordingly.

**Comment by jeryj** [comment]
@retrofox, I wasn't 100% sure I was on the right track, but I updated the `ColorPaletteControl` to have a `fieldset` with a `legend` for the title in #19546.

**Comment by retrofox** [comment]
@jeryj , I think it's the right track to take. Updating PR description. Thanks!
